lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1513,7 +1513,54 @@ convertStatusKind(core::Rule::StatusKind kind) {
   return BuildSystemDelegate::CommandStatusKind::IsScanning;
 }
 
-Rule BuildSystemEngineDelegate::lookupRule(const KeyType& keyData) {
+class BuildSystemRule : public Rule {
```

Comment:
This is a temporary bridge between the old world (std::function) and the new world.